src/libPersistence/BlockStorage.cpp

```diff
@@ -204,6 +204,56 @@ bool BlockStorage::GetRangeMicroBlocks(const uint64_t lowEpochNum,
   return true;
 }
 
+bool BlockStorage::PutTempState(const unordered_map<Address, Account>& states) {
+  // LOG_MARKER();
+
+  unordered_map<string, string> states_str;
+  for (const auto& state : states) {
+    bytes rawBytes;
+    if (!state.second.SerializeBase(rawBytes, 0)) {
+      LOG_GENERAL(WARNING, "Messenger::SetAccountBase failed");
+      continue;
+    }
+    states_str.emplace(state.first.hex(),
+                       DataConversion::CharArrayToString(rawBytes));
+  }
+
+  return m_tempStateDB->BatchInsert(states_str);
+}
+
+bool BlockStorage::GetTempStateInBatch(leveldb::Iterator*& iter,
+                                       vector<StateSharedPtr>& states) {
+  // LOG_MARKER();
+
+  lock_guard<mutex> g(m_mutexTempState);
+
+  if (iter == nullptr) {
+    iter = m_tempStateDB->GetDB()->NewIterator(leveldb::ReadOptions());
+    iter->SeekToFirst();
+  }
+
+  unsigned int counter = 0;
+
+  for (; iter->Valid() && counter < ACCOUNT_IO_BATCH_SIZE;
```

Comment:
For consistency with `RepopulateStateTrie`, maybe change to `counter <=`?

Reply:
`RepopulateStateTrie` is using `>=`, so in the case of consistency it's better to use `<`